### ISSUE TITLE
clh: vfio: virtiofs: Unexpected poll revents 11: kill event on queue 1 - quitting

### ISSUE THREAD

**Issue by devimc** [kata-containers/tests#2963]
unable to hotplug a vfio device to cloud-hypervisor

```
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000067] virtio_send_data_iov: after skip skip_size=0 in_sg_cpy_count=18 in_sg_left=73728
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000067] virtio_send_data_iov: preadv ret=73728 len=73728
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Got queue event on Queue 1
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Queue 1 gave evalue: 1 available: in: 28688 out: 80
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Waiting for Queue 1 event
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] fv_queue_worker: elem 35: with 2 out desc of length 80 bad_in_num=0 bad_out_num=0
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] unique: 54, opcode: READ (15), nodeid: 7, insize: 80, pid: 70
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] lo_read(ino=7, size=28672, off=643072)
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] virtio_send_data_iov: count=1 len=28672 iov_len=16
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] virtio_send_data_iov: elem 35: with 8 in desc of length 28688
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] virtio_send_data_iov: after skip skip_size=0 in_sg_cpy_count=7 in_sg_left=28672
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000068] virtio_send_data_iov: preadv ret=28672 len=28672
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Got queue event on Queue 1
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Queue 1 gave evalue: 1 available: in: 12448 out: 124
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Waiting for Queue 1 event
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Got queue event on Queue 1
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Queue 1 gave evalue: 1 available: in: 0 out: 0
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: Waiting for Queue 1 event
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] fv_queue_worker: elem 35: with 2 out desc of length 80 bad_in_num=0 bad_out_num=0
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] unique: 56, opcode: READ (15), nodeid: 7, insize: 80, pid: 70
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] lo_read(ino=7, size=12288, off=425984)
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] virtio_send_data_iov: count=1 len=12288 iov_len=16
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] virtio_send_data_iov: elem 35: with 4 in desc of length 12304
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] virtio_send_data_iov: after skip skip_size=0 in_sg_cpy_count=3 in_sg_left=12288
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] virtio_send_data_iov: preadv ret=12288 len=12288
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] fv_queue_worker: elem 41: with 2 out desc of length 44 bad_in_num=0 bad_out_num=0
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] unique: 58, opcode: LOOKUP (1), nodeid: 3, insize: 44, pid: 70
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] lo_lookup(parent=3, name=etc)
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070]    unique: 58, error: -2 (No such file or directory), outsize: 16
Oct 09 19:59:31 test virtiofsd[159240]: [ID: 00000070] virtio_send_msg: elem 41: with 2 in desc of length 144
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.354955044Z" level=error msg="Wait for process failed" container=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 error="ttrpc: closed" pid=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.355123018Z" level=warning msg="failed to get OOM event from sandbox" error="ttrpc: closed" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.354955044Z" level=error msg="Wait for process failed" container=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 error="ttrpc: closed" pid=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.355123018Z" level=warning msg="failed to get OOM event from sandbox" error="ttrpc: closed" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.355174216Z" level=warning msg="failed to get OOM event from sandbox" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.355174216Z" level=warning msg="failed to get OOM event from sandbox" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.788648195Z" level=error msg="createContainer failed" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=kata_agent
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.788648195Z" level=error msg="createContainer failed" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=kata_agent
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.789066072Z" level=error msg="container create failed" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.789066072Z" level=error msg="container create failed" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.799866685Z" level=warning msg="Could not umount" error="no such file or directory" host-path=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/mounts/077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d-07c56b2e5c443eba-hosts pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.799936755Z" level=error msg="rollback failed unmountHostMounts()" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.799988152Z" level=warning error="no such file or directory" pid=159224 rootfs-dir=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/mounts/077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d/rootfs sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=mount
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.799866685Z" level=warning msg="Could not umount" error="no such file or directory" host-path=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/mounts/077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d-07c56b2e5c443eba-hosts pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.799936755Z" level=error msg="rollback failed unmountHostMounts()" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.799988152Z" level=warning error="no such file or directory" pid=159224 rootfs-dir=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/mounts/077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d/rootfs sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=mount
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.818264673Z" level=warning msg="sandbox stopped unexpectedly" error="failed to ping hypervisor process: Get \"http://localhost/api/v1/vmm.ping\": context deadline exceeded" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.818264673Z" level=warning msg="sandbox stopped unexpectedly" error="failed to ping hypervisor process: Get \"http://localhost/api/v1/vmm.ping\": context deadline exceeded" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=containerd-kata-shim-v2
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.818586983Z" level=warning msg="Could not remove container share dir" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 share-dir=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test kata[159224]: time="2020-10-09T19:59:33.81892061Z" level=warning msg="Agent did not stop sandbox" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 sandboxid=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.818586983Z" level=warning msg="Could not remove container share dir" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 share-dir=/run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=container
Oct 09 19:59:33 test containerd[156800]: time="2020-10-09T19:59:33.818920610Z" level=warning msg="Agent did not stop sandbox" error="Dead agent" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 sandboxid=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000001] virtio_loop: Unexpected poll revents 11
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000001] virtio_loop: Stopping queue 0 thread
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000003] fv_queue_thread: kill event on queue 0 - quitting
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000001] virtio_loop: Stopping queue 1 thread
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000006] fv_queue_thread: kill event on queue 1 - quitting
Oct 09 19:59:33 test virtiofsd[159240]: [ID: 00000001] virtio_loop: Exit
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.610220016Z" level=info msg="shim disconnected" id=077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.610999785Z" level=error msg="Failed to pipe stdout of container \"077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d\"" error="reading from a closed fifo"
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.611078482Z" level=info msg="Finish piping stdout of container \"077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d\""
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.615380031Z" level=error msg="StartContainer for \"077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d\" failed" error="failed to create containerd task: Dead agent: unknown"
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.630424557Z" level=info msg="StopPodSandbox for \"932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4\""
Oct 09 19:59:34 test containerd[156800]: time="2020-10-09T19:59:34.630502682Z" level=info msg="Container to stop \"077c25689a8c82f2d41f03942f9b64d22fd51dc34c8db5f60d342467587b6e1d\" must be in running or unknown state, current state \"CONTAINER_EXITED\""
Oct 09 19:59:36 test containerd[156800]: time="2020-10-09T19:59:36.632303883Z" level=error msg="get state for 932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4" error="context deadline exceeded: unknown"
Oct 09 19:59:36 test containerd[156800]: time="2020-10-09T19:59:36.632441492Z" level=warning msg="unknown status" status=0
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.132673149Z" level=warning msg="VM still running after waiting 3s" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=cloudHypervisor
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.133042229Z" level=warning msg="removing vm socket failed" path=/run/vc/vm/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/clh.sock pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=cloudHypervisor
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.132673149Z" level=warning msg="VM still running after waiting 3s" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=cloudHypervisor
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.133042229Z" level=warning msg="removing vm socket failed" path=/run/vc/vm/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/clh.sock pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=cloudHypervisor
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.133960811Z" level=error msg="Failed to read guest console logs" console-protocol=pty console-url=/dev/pts/0 error="read /dev/pts/0: file already closed" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.133960811Z" level=error msg="Failed to read guest console logs" console-protocol=pty console-url=/dev/pts/0 error="read /dev/pts/0: file already closed" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.142165323Z" level=warning msg="sandbox cgroups path is empty" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.142165323Z" level=warning msg="sandbox cgroups path is empty" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.177998998Z" level=warning msg="sandbox cgroups path is empty" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test kata[159224]: time="2020-10-09T19:59:37.178277735Z" level=error msg="failed to unmount vm share path /run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/shared" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=kata_agent
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.177998998Z" level=warning msg="sandbox cgroups path is empty" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=sandbox
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.178277735Z" level=error msg="failed to unmount vm share path /run/kata-containers/shared/sandboxes/932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4/shared" error="no such file or directory" pid=159224 sandbox=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4 source=virtcontainers subsystem=kata_agent
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.180231094Z" level=info msg="TaskExit event &TaskExit{ContainerID:932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4,ID:932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4,Pid:159224,ExitStatus:0,ExitedAt:2020-10-09 19:59:33.355109209 +0000 UTC,XXX_unrecognized:[],}"
Oct 09 19:59:37 test containerd[156800]: time="2020-10-09T19:59:37.189280277Z" level=info msg="shim disconnected" id=932a1f7ffac662644754fde952ede8592175e0f5d0fd6270c0537632c46ac6e4
```

cc @likebreath

**Comment by likebreath** [comment]
@devimc Thanks for opening the issue. Would you mind sharing more about the setup and how it can be reproduced? Also, a bit of more logs from `cloud-hypervisor` would be helpful (with [full debug enabled](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#enable-full-debug), you can get logs for both kata-runtime and cloud-hypervisor with `journalctl -t kata-runtime -t kata-hypervisor` ).

**Comment by devimc** [comment]
thanks @likebreath - I'm using kata 2.0, hence `-t kata-runtime -t kata-hypervisor` are not valid identifiers

**Comment by likebreath** [comment]
/cc @jcvenegas @amshinde @egernst  Two more special configurations/workaround are required to make VFIO CI work with clh+kata 2.0 (on top of disabling DAX), as described in @devimc's commit message:

```
virtiofsd is crashing when cloud-hypervisor is running with
`virtio_fs_cache=auto` and `default_memory=2048`. Set
`virtio_fs_cache=none` and `default_memory=1024` to workaround
this bug.
```

 @jcvenegas Any insights for the  `virtio_fs_cache=none`? Also, the `1024M` memory requirement is quite strange...

**Comment by likebreath** [comment]
Fixed by workaround from #2969 .
